### ISSUE TITLE
typescript-eslint(unbound-method): "consider using an arrow function instead" but is using an arrow function

### ISSUE THREAD

**Issue by theoludwig** [oxc-project/tsgolint#611]
https://typescript-eslint.io/rules/unbound-method

```ts
  × typescript-eslint(unbound-method): Avoid referencing unbound methods which may cause unintentional scoping of `this`.
  │ If your function does not access `this`, you can annotate it with `this: void`, or consider using an arrow function instead.
    ╭─[apps/api/src/routes/animalBatch/news.ts:41:21]
 40 │       })
 41 │       .where(({ eb, or, and }) => {
    ·                     ──
 42 │         return and([
    ╰────
```

Expected: no lint error, as we can see line 41, it uses a arrow function, and the function doesn't access `this`.

**Comment by camc314** [comment]
Thanks for reporting this - do you mind creating a more detailed reproduction - it's hard to tell what's going on by that small snippet.

Also does this behaviour dffer to the eslint rule?

here's a similar case from the eslint test suite:

https://github.com/typescript-eslint/typescript-eslint/blob/997e0c005d2d80a726249cafb7cbdf4ec287aea3/packages/eslint-plugin/tests/rules/unbound-method.test.ts#L870-L884

**Comment by theoludwig** [comment]
Sorry, should have debug, and tried to reproduce it more thoroughly before opening the issue.

I tried to reproduce it, but in fact, the error is correct. The error was not there in ESLint, only in Oxlint, but seems like Oxlint caught more (correct) cases/issues than ESLint didn't catch.

My repro is there: https://github.com/theoludwig/oxlint-tsgolint-issue-135, but it's working as expected.

Thank you for trying to help, will close the issue for now, will reopen if I can still debug down to a specific oxlint issue.
